### PR TITLE
Move print-object calls to write-object

### PR DESCRIPTION
**Pull request by yitzchak**

### CODE REVIEW

File: src/lisp/kernel/clos/print.lisp [hunk 1]
```diff
@@ -280,6 +280,23 @@ printer and we should rather use MAKE-LOAD-FORM."
...
+         (write-char #\` stream)
+         (core:write-object (second l) stream))
+        (eclector.reader:unquote
+         (write-char #\, stream)
+         (core:write-object (second l) stream))
+        (eclector.reader:unquote-splicing
+         (write-string ",@" stream)
+         (core:write-object (second l) stream))
+        (otherwise
+         (call-next-method)))
```
Comment (Bike) [inline src/lisp/kernel/clos/print.lisp:297]: Is it okay for this method to be on `list`? If we're really using `print-object` pervasively, would there not be methods on `cons` and `null` that take priority? Presumably we don't actually have such methods, if this works, but that's the organization I'd imagine

File: src/lisp/kernel/clos/print.lisp [hunk 2]
```diff
@@ -332,47 +349,15 @@ printer and we should rather use MAKE-LOAD-FORM."
 ;;; ----------------------------------------------------------------------
 
 (defmethod describe-object ((obj t) (stream t))
```
Comment (Bike) [inline src/lisp/kernel/clos/print.lisp:351]: Might be better to have this method be on `standard-object` or something instead. We probably have methods for built in classes already, but if not things could get a little weird using `class-slots` and all

File: src/core/object.cc [hunk 3]
```diff
@@ -492,11 +492,7 @@ CL_DEFUN bool core__sl_boundp(T_sp obj) {
 void General_O::describe(T_sp stream) { clasp_write_string(this->__repr__(), stream); }
 
 void General_O::__write__(T_sp strm) const {
-  if (_sym_STARliteral_print_objectSTAR->symbolValue().notnilp()) {
```
Comment (Bike) [inline src/core/object.cc:495]: as discussed on IRC - if you don't mind nuking this "literal print object" thing too, go ahead and do that

File: src/lisp/kernel/clos/print.lisp (hunk 2 above)
Comment (yitzchak) [reply src/lisp/kernel/clos/print.lisp:351]: Sounds good. BTW, we don't really have any describe-object methods right now. Most all is handled in the pre-CLOS inspect.lisp

**Review by Bike — APPROVED** [review]

**Pull request merged**